glpi_api.py

```diff
@@ -749,10 +778,41 @@ def update(self, itemtype, *items):
                                     json={'input': items})
         return {
             200: lambda r: r.json(),
+            201: lambda r: r.json(),
             207: lambda r: r.json()[1],
             400: _glpi_error,
             401: _glpi_error
         }.get(response.status_code, _unknown_error)(response)
+    
+    @_catch_errors
+    def update_sub_items(self, itemtype, item_id, sub_itemtype, *items):
+        """`API documentation
+        Same method used as get-sub-items, same parameters 
+        <https://github.com/glpi-project/glpi/blob/master/apirest.md#get-sub-items>`__
+
+        Updates a collection of rows of the ``sub_itemtype`` for the identified
+        item of type ``itemtype`` and id ``item_id``. ``kwargs`` contains
+        additional parameters allowed by the API.
+
+        .. code::
+
+            # update the operatingsystem a computer.
+            >>> In [241]: glpi.update_sub_items('Computer',1,'Item_OperatingSystem' {'id': 1 ,'itemtype':'Computer','operatingsystem_id':1 })
+            [{'id': 261,
+              'itemtype': 'Computer',
+              'items_id': 1,
+            ...
+        """
+        url = self._set_method(itemtype, item_id, sub_itemtype)
+        response = self.session.put(url,
+                                    json={'input': items})
+        return {
+            200: lambda r: lambda r: r.json(),
+            201: lambda r: lambda r: r.json(),
+            207: lambda r: lambda r: r.json(),
+            400: lambda r: _glpi_error,
```

Comment:
no need to use a lambda if a function is returned:

`400: _glpi_error`

(and with using `lambda`, it should be `400: lambda r: _glpi_error(r)` otherwise the lambda function reference is returned and the exception from `_glpi_error` is not raised)

Reply:
Good point, corrected